### PR TITLE
SDK after_error _response

### PR DESCRIPTION
**Pull request by eeshansim**

### CODE REVIEW

File: developing-connectors/sdk/error-handling.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+This is a Workato SDK specific method to raise a job error with a custom message.
+
+```ruby
+error("Some error message")
+```
+
+This can be used in a number of ways to improve usability for your custom connector.
+
+### Input validation
+You can raise custom errors for inputs that violates some business logic. Let's use an example of a search action that tries to search for a **Contact** using a search action. You want to make sure that the user is searching for records with at least one search criteria. In this case, you will want to raise an error if there are no input values, instead of sending a request with no query parameters.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: You can raise custom errors for inputs that violate~s~

File: developing-connectors/sdk/error-handling.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+end
+```
+
+![Input validation error message](/assets/images/sdk/input-validation-error-message.png)
+*Input validation error message*
+
+### Response validation
+Some APIs do not use response codes to indicate an error with a request. These APIs may respond with `200` HTTP code and error messages in the response body. **Sage Intacct** is a good example.
+
+When there are [Sage Intacct business logic errors](https://developer.intacct.com/web-services/error-handling#business-logic-errors) with a HTTP request, the response from Intacct Web Service will also be a `200` response code with an `<errormessage>` XML tag in the respond body.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: ~respond~ response body

File: developing-connectors/sdk/error-handling.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        }
+      ]
+    }
+  ]
+}
+```
+
+*If this XML-to-hash conversion is unfamiliar to you, refer to the documentation on [XML format](/developing-connectors/sdk/data-format/xml-format.md) for more information.*
+
+Hence, it is important to be able to validate response and raise an error when one is present. By default, the SDK framework will only raise errors when the HTTP response code indicates an error, such "errors" will be missed.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: > By default, the SDK framework will only raise errors when the HTTP response code indicates an error, such "errors" will be missed.

Are you trying to say that user needs to validate the response and raise an error in cases where the response is a status 200? Suggestion:

By default, the SDK framework will only raise errors when the HTTP response code indicates an error. Hence, it is important to validate all responses and raise an error if one is present in the response body.

File: developing-connectors/sdk/error-handling.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+There are 2 ways you catch these errors. The first method is to use [detect_on](/developing-connectors/sdk/authentication/custom-authentication.md#detect-on) in the `connection` definition.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: you **can** catch these errors.

File: developing-connectors/sdk/error-handling.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      error(result.dig("errormessage", 0))
+    else
+      result["data"]
+    end
+  end
+```
+
+## `after_error_response`
+
+`after_error_response` is a helper method that can be chained to a HTTP verb method to handle HTTP responses. In particular, when the API responses with an error response code.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: `after_error_response` is a helper method that can be chained to a HTTP verb method to handle HTTP responses; in particular, API responses with an error response code.

File: developing-connectors/sdk/error-handling.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+This method accepts 2 arguments. First, a number which represents the exact error code that you wish to handle.
+
+Next, it also accepts a conditional block that will be executed when a HTTP response code matching the first argument is received.
+
+Let's take a look at an example, using **Airtable** API.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: Let's take a look at an `after_error_response` example, using **Airtable** API.

File: developing-connectors/sdk/error-handling.md (hunk 4 above)
Comment (jandonyada) [reply developing-connectors/sdk/error-handling.md]: In practice, why would a user choose one over the other? Or is it purely preference? Even then, we should probably give some info.

File: developing-connectors/sdk/methods.md [hunk 7]
```diff
@@ -75,7 +75,7 @@ Learn how to make HTTP requests using the HTTP verb methods [here](/developing-c
       <td>inject</td>
       <td>
         Combine elements in an array using an operation.<br>
-        <a href="http://apidock.com/ruby/Enumerable/inject" target="_blank">More details here</a>
+        <a href="http://apidock.com/ruby/Enumerable/inject">More details here</a>
```
Comment (jandonyada) [inline developing-connectors/sdk/methods.md]: More details here.

(add full-stop)

File: developing-connectors/sdk/methods.md [hunk 8]
```diff
@@ -112,35 +112,42 @@ Learn how to make HTTP requests using the HTTP verb methods [here](/developing-c
...
+        raise a job error with a user-defined error body.<br>
+        <code>error("Unable to find Account with ID: 123")</code></a>
       </td>
     </tr>
     <tr>
       <td>reject</td>
       <td>
         Selectively returns elements for which the block returns false. Similar but opposite of <b>select</b>.<br>
-        <a href="http://apidock.com/ruby/v1_9_3_392/Array/reject" target="_blank">More details here</a>
+        <a href="http://apidock.com/ruby/v1_9_3_392/Array/reject">More details here</a>
```
Comment (jandonyada) [inline developing-connectors/sdk/methods.md]: Same here, full-stop at end of sentence.

Comment (jandonyada) [reply developing-connectors/sdk/methods.md]: Same for the rest of the links.

File: developing-connectors/sdk/error-handling.md (hunk 3 above)
Comment (eeshansim) [reply developing-connectors/sdk/error-handling.md]: Yes, only for certain APIs. This is much clearer, thanks.

File: developing-connectors/sdk/error-handling.md (hunk 4 above)
Comment (eeshansim) [reply developing-connectors/sdk/error-handling.md]: That's a good point. `detect_on` is used as a connector-wide method while `after_error_response` is a per action method. `after_error_response` also allows a user to customize the error message.

File: developing-connectors/sdk/error-handling.md [hunk 9]
```diff
@@ -31,7 +31,7 @@ end
 ### Response validation
 Some APIs do not use response codes to indicate an error with a request. These APIs may respond with `200` HTTP code and error messages in the response body. **Sage Intacct** is a good example.
 
-When there are [Sage Intacct business logic errors](https://developer.intacct.com/web-services/error-handling#business-logic-errors) with a HTTP request, the response from Intacct Web Service will also be a `200` response code with an `<errormessage>` XML tag in the respond body.
+When there are [Sage Intacct business logic errors](https://developer.intacct.com/web-services/error-handling#business-logic-errors) with a HTTP request, the response from Intacct Web Service will also be a `200` response code with an `<errormessage>` XML tag in the response body.
 
 For example, if the
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: Hmm if you continue from the examples following this, this sentence doesn't actually end anywhere.

File: developing-connectors/sdk/error-handling.md [hunk 10]
```diff
@@ -101,11 +101,11 @@ which will be transformed into a hash equivalent
...
+There are 2 ways you can catch these errors. The first method is to use [detect_on](/developing-connectors/sdk/authentication/custom-authentication.md#detect-on) in the `connection` definition. This is a connector-wide method to catch errors; It applies to all actions and triggers. Furthermore, this method will raise errors with a standard message format that cannot be customized.
 
-The alternative is to handle these errors in the [after_response](/developing-connectors/sdk/request.md#after_response) method. This method can be used together with `error()` to validate the contents of a HTTP response and raise custom errors.
+The alternative is to handle these errors in the [after_response](/developing-connectors/sdk/request.md#after_response) method. This method can be used together with `error()` to validate the contents of a HTTP response and raise custom errors. This method applies to individual action/trigger. Hence the condition(s) used to detect an error can be customized to each request. Additionally, the error message can be changed to suit each action/trigger.
```
Comment (jandonyada) [inline developing-connectors/sdk/error-handling.md]: This method applies individual action**s**/trigger**s**

File: developing-connectors/sdk/error-handling.md (hunk 9 above)
Comment (eeshansim) [reply developing-connectors/sdk/error-handling.md]: Typo

**Pull request merged**